src/queue.jl

```diff
@@ -20,10 +20,19 @@ end
 
 dequeue!(s::Queue) = shift!(s.store)
 
-function iter{T}(q::Queue{T})
+# Iterators
+
+start(q::Queue) = start(q.store)
+next(q::Queue, s) = next(q.store, s)
+done(q::Queue, s) = done(q.store, s)
+
+reverse_iter(q::Queue) = reverse_iter(q.store)
+
+
+function iter{T}(s::Stack{T})
     a = T[]
-    for i in q.store
-        push!(a,i)
+    for i in s.store
+        unshift!(a,i)
```

Comment:
I think you forgot to remove this function.  (Also not sure why the type was changed to `Stack`.)